src/miral/basic_window_manager.cpp

```diff
@@ -2902,15 +2932,25 @@ void miral::BasicWindowManager::update_application_zones_and_attached_windows()
             if (window)
             {
                 auto& info = info_for(window);
+                if (surface_stack->screen_is_locked() && !info.visible_on_lock_screen())
+                    continue;
 
                 switch (info.state())
                 {
                 case mir_window_state_attached:
-                    if (info.exclusive_rect().is_set())
+                    if (info.ignore_exclusion_zones())
+                    {
+                        full_screen_attached_windows.push_back(&info);
```

Comment:
`ignore_exclusion_zones()` does not imply fullscreen. Just that to ignores exclusion zones for attached edges.